cmd/monitor/monitor.go

```diff
@@ -102,6 +101,66 @@ func getChainState(ctx context.Context, ec *ethclient.Client) (*chainState, erro
 
 }
 
+func fetchBlocks(ctx context.Context, ec *ethclient.Client, ms *monitorStatus, rpc *ethrpc.Client, isUiRendered bool) (err error) {
+	var cs *chainState
+	cs, err = getChainState(ctx, ec)
+	if err != nil {
+		log.Error().Err(err).Msg("Encountered issue fetching network information")
+		time.Sleep(interval)
+		// continue
+		return
+	}
+
+	batchSize := 100
+	if isUiRendered {
+		_, termHeight := ui.TerminalDimensions()
+		batchSize = termHeight/2 - 4
+	}
```

Comment:
Ah, using the terminal dimensions is a good idea! I think not having a `batch-size` flag or limiting the number of blocks requested could cause issues with some RPC providers (as they have limits).

Additionally, when testing with `go run main.go monitor https://polygon-rpc.com` I noticed that scrolling down was really laggy, scrolling up was "okay". I think the previous implementation of `window-size` remedies some of this. Essentially its purpose was to only render the blocks the were visible in the terminal window.

Reply:
👍 i'll have to test with `https://polygon-rpc.com`. ideally, i think `batch-size` and `window-size` should just go away. i think it's just hacks to manually set the sizing, am i seeing this right? wyt?

Reply:
I think window-size can go, but batch size should stay, since it'll determine the max batch requests at once. Some RPC providers have limits, so making it configurable would be good.